doc/develop/modules.rst

```diff
@@ -131,17 +166,24 @@ changes that are to be brought into the module repository.
 
 .. note::
 
-     Force-pushing to a module's main branch is not allowed.
+     Force-pushing to a module's main branch is not allowed unless it is necessary to
+     revert an incorrect previous push which forked the branch from the upstream.
 
 Allowed practices
 -----------------
 
 The following practices conform to the above requirements and should be
-followed in all modules repositories. It is up to the module code owner
-to select the preferred synchronization practice, however, it is required
-that the selected practice is consistently followed in the respective
+followed in modules repositories which keep downstream Zephyr specific changes.
+It is up to the module code owner to select the preferred synchronization practice,
+however, it is required that the selected practice is consistently followed in the respective
 module repository.
 
+**Updating by rebasing:**
```

Comment:
The MCUboot currently uses strategy of fast-forward push to main. This is done via command line, but allows us to keep the MCUboot main branch with the same history (sha) as the upstream.